classes/local/util/manager.php

```diff
@@ -154,6 +154,41 @@ protected function prepare_files_skeletons() {
                 $this->prepare_file_skeleton('db/upgradelib.php', 'php_internal_file', 'db_upgradelib');
             }
         }
+
+        if ($this->should_have('backup_moodle2')) {
+            $this->prepare_backup_moodle2();
+        }
+    }
+
+    /**
+     * Prepares the skeleton files for the 'backup_moodle2' feature.
+     */
+    public function prepare_backup_moodle2() {
+
+        // TODO: Lib.php will advertise the fact that the plugin supports backup.
```

Comment:
There we're just adding the file lib.php to the list of files to be created.

In the lib_php_file class there's a check for the 'backup_moodle2' feature. If the feature is present, then FEATURE_BACKUP_MOODLE2 is added to the list of supported features.

In effect, it's still the lib_php_file that is adding content as appropiate, and only to the recipe for lib.php, in the manager we're just queuing the file to be created.